Src/WindowsServer/WindowsServer.Shared/Implementation/AzureMetadataRequestor.cs

```diff
@@ -0,0 +1,113 @@
+﻿namespace Microsoft.ApplicationInsights.WindowsServer.Implementation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Net;
+    using System.Net.Http;
+    using System.Runtime.Serialization.Json;
+    using System.Threading.Tasks;
+#if NETSTANDARD1_3
+    using System.Net.Http;
+#endif
+    using Extensibility;
+    using Microsoft.ApplicationInsights.WindowsServer.Implementation.DataContracts;
+
+    internal class AzureMetadataRequestor : IAzureMetadataRequestor
+    {
+        /// <summary>
+        /// Azure Instance Metadata Service exists on a single non-routable IP on machines configured
+        /// by the Azure Resource Manager. See <a href="https://go.microsoft.com/fwlink/?linkid=864683">to learn more.</a>
+        /// </summary>
+        internal static string AzureImsApiVersion = $"api-version=2017-08-01"; // this version has the format=text capability
+        internal static string AzureImsJsonFormat = "format=json";
+        internal static int AzureImsMaxResponseBufferSize = 512;
+
+        /// <summary>
+        /// Default timeout for the web requests made to obtain Azure IMS data. Internal to expose to tests.
+        /// </summary>
+        internal TimeSpan AzureImsRequestTimeout = TimeSpan.FromSeconds(10);
+
+        /// <summary>
+        /// Private function for mocking out the actual call to IMS in unit tests. Available to internal only.
+        /// </summary>
+        private Func<string, Task<AzureInstanceComputeMetadata>> azureIMSRequestor = null;
+
+        private string baseImdsUri = $"http://169.254.169.254/metadata/instance/compute";
+
+        internal AzureMetadataRequestor(Func<string, Task<AzureInstanceComputeMetadata>> makeAzureIMSRequestor = null)
```

Comment:
What is the string parameter here? Is it used anywhere?

Reply:
That’s up to the caller to decide. For instance, if we point this functor at ‘MakeWebRequest’ the string value is the Uri to call to obtain the JSON response. This method will be used only for Test scenarios. I’ve updated the docs to reflect this.